sklearn/covariance/outlier_detection.py

```diff
@@ -62,9 +69,16 @@ def decision_function(self, X, raw_values=False):
             such as the One-Class SVM.
 
         """
+
+        if raw_values is not None:
+            warnings.warn(
+                'raw_values keyword has been moved to class initialization',
+                DeprecationWarning)
+            self.raw_values = raw_values
+
         check_is_fitted(self, 'threshold_')
         mahal_dist = self.mahalanobis(X)
-        if raw_values:
+        if self.raw_values:
             decision = mahal_dist
         else:
```

Comment:
@jnothman I agree but how would we do that? If it is set in ``__init__`` it will return the negative ones and otherwise not? good with me but we must document it carefully